ext/standard/basic_functions.c

```diff
@@ -4662,14 +4662,16 @@ PHP_FUNCTION(time_sleep_until)
 	/* 1sec = 1000000000 nanoseconds */
 	php_req.tv_nsec = (long) ((c_ts - php_req.tv_sec) * 1000000000.00);
```

Comment:
All of this double based code looks super fishy, especially the "rounding up occurred" branch above. Unfortunately the function accepts a double argument, ugh.

Reply:
I ... didn't even look at these parts, assumption is the mother of all ... and so on ...

What would be your suggestion ?